src/classes/RecipeRepository.js

```diff
@@ -41,16 +42,15 @@ class RecipeRepository {
       recipe.ingredients.find(ingredient => ingredient.name === this.filterTerm);
     }) : this.tag !== '' && this.filterTerm !== '' ?
     this.currentRecipes = this.currentRecipes.filter(recipe => {
-      return recipe.tags.includes(this.tag) ||
-      recipe.name.toLowerCase().includes(this.filterTerm) ||
-      recipe.ingredients.find(ingredient => ingredient.name === this.filterTerm)
+      return recipe.tags.includes(this.tag) &&
+      (recipe.name.toLowerCase().includes(this.filterTerm) ||
+      recipe.ingredients.find(ingredient => ingredient.name === this.filterTerm))
```

Comment:
This is how we fixed the logic to filter for both terms at once, instead of either/or